src/herder/HerderSCPDriver.cpp

```diff
@@ -1223,25 +1223,15 @@ HerderSCPDriver::checkAndCacheTxSetValid(TxSetXDRFrame const& txSet,
     bool* pRes = mTxSetValidCache.maybeGet(key);
     if (pRes == nullptr)
     {
-        // The invariant here is that we only validate tx sets nominated
-        // to be applied to the current ledger state. However, in case
-        // if we receive a bad SCP value for the current state, we still
-        // might end up with malformed tx set that doesn't refer to the
-        // LCL.
-        ApplicableTxSetFrameConstPtr applicableTxSet;
-        if (txSet.previousLedgerHash() ==
-            mApp.getLedgerManager().getLastClosedLedgerHeader().hash)
-        {
-            applicableTxSet = txSet.prepareForApply(mApp);
-        }
-
-        bool res = true;
+        bool res;
+        auto applicableTxSet = txSet.prepareForApply(mApp);
```

Comment:
prepareForApply asserts that previous ledger hash equals LCL, so I don't think we can make this change without reworking `prepareForApply` as well.